src/lib/utils/package-manifest.ts

```diff
@@ -0,0 +1,193 @@
+// Utilities to support the inspection of node package "manifests" (package.json's)
+
+import { existsSync, statSync } from "fs";
+import glob = require("glob");
+import { dirname, resolve } from "path";
+import { flatMap } from "./array";
+
+import { readFile } from "./fs";
+import { Logger } from "./loggers";
+
+/**
+ * Helper for the TS type system to understand hasOwnProperty
+ * and narrow a type appropriately.
+ * @param obj the receiver of the hasOwnProperty method call
+ * @param prop the property to test for
+ */
+function hasOwnProperty<X extends {}, Y extends PropertyKey>(
+    obj: X,
+    prop: Y
+): obj is X & Record<Y, unknown> {
+    // eslint-disable-next-line no-prototype-builtins
+    return obj.hasOwnProperty(prop);
+}
+
+/**
+ * Loads a package.json and validates that it is a JSON Object
+ */
+export function loadPackageManifest(
+    logger: Logger,
+    packageJsonPath: string
+): Record<string, unknown> | undefined {
+    const packageJson: unknown = JSON.parse(readFile(packageJsonPath));
+    if (typeof packageJson !== "object" || !packageJson) {
+        logger.error(`The file ${packageJsonPath} is not an object.`);
+        return undefined;
+    }
+    return packageJson as Record<string, unknown>;
+}
+
+/**
+ * Load the paths to packages specified in a Yarn workspace package JSON
+ * Returns undefined if packageJSON does not define a Yarn workspace
+ * @param packageJSON the package json object
+ */
+function getPackagePaths(
+    packageJSON: Record<string, unknown>
+): string[] | undefined {
+    if (Array.isArray(packageJSON.workspaces)) {
```

Comment:
We're missing a check that the array is an array of strings - `Array.isArray` gives us `any[]`

Reply:
Good catch.